android/java/org/chromium/chrome/browser/crypto_wallet/fragments/ApproveTxBottomSheetDialogFragment.java

```diff
@@ -211,32 +212,26 @@ public void setupDialog(@NonNull Dialog dialog, int style) {
         assert jsonRpcService != null && keyringService != null;
 
         TextView networkName = view.findViewById(R.id.network_name);
-        TextView txType = view.findViewById(R.id.tx_type);
 
         mCoinType = TransactionUtils.getCoinFromTxDataUnion(mTxInfo.txDataUnion);
         jsonRpcService.getNetwork(mCoinType, selectedNetwork -> {
             networkName.setText(selectedNetwork.chainName);
             keyringService.getKeyringInfo(Utils.getKeyringForCoinType(mCoinType), keyringInfo -> {
                 final AccountInfo[] accounts = keyringInfo.accountInfos;
+                // First fill in data that does not require remote queries
+                TokenUtils.getAllTokensFiltered(getBraveWalletService(), getBlockchainRegistry(),
+                        selectedNetwork, selectedNetwork.coin, TokenUtils.TokenType.ALL,
+                        fullTokenList -> {
+                            fillAssetDependentControls(view, selectedNetwork, accounts,
```

Comment:
why do we call `fillAssetDependentControl` twice? It could be that it's called in the same moment after `TokenUtils.getAllTokensFiltered` and `Utils.getTxExtraInfo`. Doesn't it bring an extra overhead? We call it once with empty prices, but second time with all prices. I think btw the main problem is that we pull all tokens asset prices there. And that's why it takes so long.